### PR TITLE
Initial CI

### PR DESCRIPTION
**Pull request by brainstorm**
Should fix issue #9.

### CODE REVIEW

**Comment by chris-zen** [comment]
As far as I see there are two problems in Travis:
- When compiling for linux, it complains about a dependency exclusive for osx
  This can be solved restricting the coremidi dependency to osx only. See [this](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies).
- When compiling for osx, it misses the native library for portaudio
   In the travis `before_install` section you need to install portaudio. Please look at the [README.md](https://github.com/chris-zen/hero-studio/blob/audio-lockfree/README.md#installing-portaudio--portmidi-in-ubuntu) for instructions on how to do it (not sure if it will be so trivial in travis).

Please let me know if you want any help with them.

**Comment by brainstorm** [comment]
The second point seems to be an easy fix with the TravisCI addons (Homebrew)... the first one, I would go for matrix-specific targets like ripgrep does instead of fiddling with one of the multiple Cargo.toml you have:

https://github.com/BurntSushi/ripgrep/blob/master/.travis.yml

Thoughts?

**Comment by chris-zen** [comment]
In this project you still need to follow [this](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies) and specify dependencies depending on the target os, because it contains [conditional code depending on it](https://github.com/chris-zen/hero-studio/blob/audio-lockfree/app-server/src/midi/drivers/mod.rs#L4).

It affects to `app-server`, and it would be something like (remember to remove coremidi from the general section):
```toml
[target.'cfg(target_os = "macos")'.dependencies]
coremidi = "0.3.1"
```

File: app-server/Cargo.toml [hunk 1]
```diff
@@ -15,11 +15,13 @@ failure_derive = "*"
...
 
 [dependencies.websocket]
 version = "0.22.2"
 default-features = false
-features = ["sync-ssl"]
\ No newline at end of file
+features = ["sync-ssl"]
+
+[target.'cfg(target_os = "macos")'.dependencies]
+coremidi = { git = "https://github.com/chris-zen/coremidi.git", branch = "packet-buffer-with-capacity" }
```
Comment (chris-zen) [inline app-server/Cargo.toml]: @brainstorm I already merged and released that branch into crates.io with the version `0.3.1`. Would you mind changing it to `coremidi = "0.3.1"` please ?

**Comment by chris-zen** [comment]
@brainstorm this looks great to me. Let me know if you need any help to fix the failing tests ;-)

**Comment by chris-zen** [comment]
Thanks @brainstorm for the great work !!! I realised how much in need I was for this PR after all the problems surfaced by it.
No more cats will die from now on 😜

**Pull request merged**